first draft of hanee's post

nitbix opened the pull request.

Commit:
first draft of hanee's post

Commit:
add missing images

Commit:
new code version

aturley committed:
Set draft to false

Setting draft to false so that the post gets rendered in preview.

aturley committed:
Change Hanee's profile file name

The file name was different that what was being used in the slug for
the blog post, which cause Hugo to spit out an error, which cause the
netlify preview to fail. This change keeps things from failing.

Commit:
fix headers

Commit:
fix headers

JONBRWN committed:
Furter updates to Hanee's blog post

- Adds link to project on Github.
- Updates code snippets to match the latest code.
- Adds Python dependencies.
- Adds PYTHONPATH setup instructions.
- Adds recommendation to use virtualenv if running into issues.

SeanTAllen commented on content/post/twitter-hashtags-real-time.md:
```diff
@@ -0,0 +1,562 @@
+
+Now let’s define the computation class which will be used in the data processing logic. We will be applying this on the tweets received from the Decoder in order to extract all the Hashtags.
+
+Our computation is called ** HashtagFinder**, and like all computations, it must implement the following two methods:
```
I think you want **HashtagFinder**

chuckblake committed:
Style and link fixes

nisanharamati commented on content/post/twitter-hashtags-real-time.md:
```diff
@@ -0,0 +1,561 @@
+
+First, a little background on Wallaroo.  Wallaroo is a relatively new open-source project that has been getting a lot of attention recently.  Wallaroo Labs, the company behind the project, has shared a good deal of information about their approach and technology. You may have seen some of their recent blog articles on Hacker News.
+
+You can find more information about Wallaroo by visiting [this site](wallaroolabs.com/community).
```
This href needs the `https://` at the beginning otherwise it assumes it's a relative path to the blog post uri

aturley replied:
Fixed.

nisanharamati commented on content/post/twitter-hashtags-real-time.md:
```diff
@@ -0,0 +1,561 @@
+
+Let's start by creating a client that connects to the Twitter API in order to grab the tweets and send them to Wallaroo.
+
+**twitter_client.py** connects to Wallaroo using a TCP connection, calls the Twitter Streaming API to get the tweets in real-time, and forwards them to Wallaroo in order to be processed.
```
`**...**` bold doesn't seem to render on our blog. You should use code tags instead

aturley replied:
Fixed.

nisanharamati commented on content/post/twitter-hashtags-real-time.md:
```diff
@@ -0,0 +1,561 @@
+
+**twitter_client.py** connects to Wallaroo using a TCP connection, calls the Twitter Streaming API to get the tweets in real-time, and forwards them to Wallaroo in order to be processed.
+
+We connect to our Wallaroo application via a socket on **('localhost', 8002)**, then call **get_tweets()**, which initiates the connection to Twitter and forwards the tweets to Wallaroo, with the help of **send\_tweets\_to\_wallaroo(http\_resp, tcp\_connection)**.
```
replace `** bold **` tags with code tags

aturley replied:
Fixed.

nisanharamati commented on content/post/twitter-hashtags-real-time.md:
```diff
@@ -0,0 +1,561 @@
+
+We connect to our Wallaroo application via a socket on **('localhost', 8002)**, then call **get_tweets()**, which initiates the connection to Twitter and forwards the tweets to Wallaroo, with the help of **send\_tweets\_to\_wallaroo(http\_resp, tcp\_connection)**.
+
+Note that in **send\_tweets\_to\_wallaroo(http\_resp, tcp\_connection)** we compute **payload_length** for each tweet and format it as a **5-digit** string (e.g. **'00005'**), which is sent along with the tweet.
```
Replace `** bold **` with code tags

aturley replied:
Fixed.

nisanharamati commented on content/post/twitter-hashtags-real-time.md:
````diff
@@ -0,0 +1,561 @@
+Before we run our Twitter Trending Hashtags application, we need to make sure we have the proper Python dependencies installed. We depend on `pandas`, `requests_oauthlib`, and `flask`. These can be installed with the following commands:
+
+```
+pip install pandas requests_oauthlib flask
````
This installation takes a long time, and could be moved to an earlier point in the article, so it runs while the user is setting up other things, like the twitter app auth tokens

nisanharamati replied:
Also this needed to be run with sudo if not run in a virtualenv (on the Ubuntu Trusty VM I was using)

nisanharamati commented on content/post/twitter-hashtags-real-time.md:
```diff
@@ -0,0 +1,561 @@
+
+Now we can build the Wallaroo application that identifies the trending hashtags on the real-time stream.
+
+The Wallaroo application logic is self-contained in **twitter\_wallaroo\_app.py**. We start by importing all the needed libraries.
```
replace `**bold**` with backticks

aturley replied:
Fixed.

nisanharamati commented on content/post/twitter-hashtags-real-time.md:
```diff
@@ -0,0 +1,561 @@
+
+### 5. Create The Decoder
+
+The [Decoder](https://docs.wallaroolabs.com/book/core-concepts/core-concepts.html) will translate the raw messages from the network connection and feed the resulting messages to the computations. Create a class called **Decoder** that implements the following three methods.
```
replace `** **` with backticks

aturley replied:
Fixed.

nisanharamati commented on content/post/twitter-hashtags-real-time.md:
```diff
@@ -0,0 +1,561 @@
+
+The [Decoder](https://docs.wallaroolabs.com/book/core-concepts/core-concepts.html) will translate the raw messages from the network connection and feed the resulting messages to the computations. Create a class called **Decoder** that implements the following three methods.
+
+* **header_length(self)**: This method returns a fixed integer that represents the number of bytes that hold the value of **payload_length**. In this case we return 5, which denotes that the value of **payload_length** is held in 5 bytes.
```
replace `** **` with backticks

aturley replied:
Fixed.

nisanharamati commented on content/post/twitter-hashtags-real-time.md:
```diff
@@ -0,0 +1,561 @@
+Our computation is called **HashtagFinder**, and like all computations, it must implement the following two methods:
+
+* **name(self)**:  returns the name of the computation step.
+* **compute(self, input_data)**: takes a parameter **input_data**, containing the data coming from the previous step and implements the current step’s logic.
```
The code below uses `compute_multi`, so maybe this part should use the same?

aturley committed:
Replace double-starts with backticks in the Twitter Hashtag post

Parts of the blog post use double-stars to surround code, which is
rendered as bold in some markdown engines. Hugo doesn't support the
use of "**" for bold, and our standard style is to use backticks for
code, so this change makes the text display correctly and consistently
with our other posts.

nisanharamati commented on content/post/twitter-hashtags-real-time.md:
````diff
@@ -0,0 +1,561 @@
+
+The last part is the function that repeats an Ajax request every second to **/refreshDashboard**, which returns the updated data for the chart.
+
+```javascript
````
It's not clear where this code is supposed to go.

nisanharamati commented on content/post/twitter-hashtags-real-time.md:
```diff
@@ -0,0 +1,561 @@
+
+Now that we have built all the components, from grabbing the data all the way to representing it on a dashboard, the only remaining step it to run everything:
+
+1. Run the Dashboard application: **app.py**
```
Include command line to run

nisanharamati commented on content/post/twitter-hashtags-real-time.md:
```diff
@@ -0,0 +1,561 @@
+Now that we have built all the components, from grabbing the data all the way to representing it on a dashboard, the only remaining step it to run everything:
+
+1. Run the Dashboard application: **app.py**
+2. Run **socket\_receiver.py**
```
include command line to run

nisanharamati commented on content/post/twitter-hashtags-real-time.md:
```diff
@@ -0,0 +1,561 @@
+2. Run **socket\_receiver.py**
+3. Run the **twitter\_wallaroo\_app** from terminal using the below commands:
+
+Note: The `machida` executable is in `machida/build/machida` in the [Wallaroo](https://github.com/wallaroolabs/wallaroo) repo. For example, if you’ve followed the Wallaroo installation [instructions](https://docs.wallaroolabs.com/book/getting-started/setup.html) then it will be in `$HOME/wallaroo-tutorial/wallaroo/machida/build/machida`. In order to run `machida` you will need to set up your `PYT ...
```
Is this supposed to be indented into the bullet point?

nisanharamati commented on content/post/twitter-hashtags-real-time.md:
````diff
@@ -0,0 +1,561 @@
+  --ponynoblock
+```
+
+4.  Run **twitter_client.py**
````
Since the indentation above is broken, this appears as a new numbered bullet point (starting from 1)

aturley committed:
Fix link to community page in Twitter Hashtag post

The URL did not begin with https:// so it was treated as a relative
URL and did not point to an actual endpoint.

nisanharamati committed:
Fix link to community page

nisanharamati commented on content/post/twitter-hashtags-real-time.md:
```diff
@@ -0,0 +1,560 @@
+Our computation is called `HashtagFinder`, and like all computations, it must implement the following two methods:
+
+* `name(self)`:  returns the name of the computation step.
+* `compute(self, input_data)`: takes a parameter `input_data`, containing the data coming from the previous step and implements the current step’s logic.
```
Needs an explanation for why this is `compute` but the code below is `compute_multi`

nisanharamati committed:
Remove spurious `\`

nisanharamati committed:
Remove more instances of `\_`

nisanharamati committed:
Add run commands and fix indents

aturley committed:
Move pip instructions, change compute method description

The pip instructions were at the end of the blog post. They take a
long time to complete, so they were moved to the beginning so that the
user can continue with other instructions while the packages are
downloading.

There was a section that described the compute method, but did not
describe compute_multi, which was used by the class below it. The
description has been updated to describe both compute and
compute_multi.

Changed formatting of some of the headers because they used numbers,
which is inconsistent with our other posts and didn't really add
anything to the readability of the post.

aturley committed:
Add "python" tag to Twitter hashtag post

I added the "python" tag to the Twitter Hashtag blog post because that
will cause it to get picked up by planetpython.org.

SeanTAllen committed:
Fix Hanne' author image
